### PR TITLE
Complete Elasticsearch logger names

### LINKED ISSUE #20326: Elasticsearch logger names
When using Log4j and other Java-based loggers, the name of a logger tied to a class is the fully-qualified name of the class. For Elasticsearch, this would be `org.elasticsearch...` (e.g. the logger tied to `Bootstrap` would be `org.elasticsearch.bootstrap.Bootstrap`). However, Elasticsearch has long had a feature that drops the `org.elasticsearch` (so the logger for `Bootstrap` becomes `bootstrap.Bootstrap`).

I suppose this is just to remove some verbosity, but it is inconsistent with logger names in other projects, and even inconsistent with other logger names that appear in our plugins (for example, in the AWS SDK based plugins we do not drop `com.amazonaws`). We do modify `org.jboss.netty` to `netty3` and `io.netty` to just `netty`, but we do not do this for any other loggers that get exposed.

It causes another problem though: there is no way to set the logger level for only the Elasticsearch packages. That is, if you want to set the logger level to trace for Elasticsearch, then this means setting the root logger level which means picking up everything else (e.g., Netty) even if you don't want that.

The question here is, should we remove this feature and just always require the `org.elasticsearch` prefix? Note that this should not impact the ability to drop the `org.elasticsearch` prefix when the package name is logged as part of the log format, we can control that separately.

### PR DESCRIPTION
**Pull request by jasontedor**
This commit modifies the logger names within Elasticsearch to be the
fully-qualified class name as opposed removing the org.elasticsearch
prefix and dropping the class name. This change separates the root
logger from the Elasticsearch loggers (they were equated from the
removal of the org.elasticsearch prefix) and enables log levels to be
set at the class level (instead of the package level).

Closes #20326

### CODE REVIEW

**Comment by jasontedor** [comment]
retest this please

**Comment by abeyad** [comment]
LGTM

**Comment by jasontedor** [comment]
retest this please

**Comment by jasontedor** [comment]
This pull request unearthed an issue that exists in master today regarding the hierarchy of setting logger level. I opened #20463.

**Comment by abeyad** [comment]
@jasontedor the Lucene logger name commit LGTM

**Comment by jasontedor** [comment]
Thank you @abeyad.

File: core/src/main/java/org/elasticsearch/common/lucene/LoggerInfoStream.java [hunk 1]
```diff
@@ -53,14 +46,11 @@ public boolean isEnabled(String component) {
     }
 
     private Logger getLogger(String component) {
-        if (component.equals("IFD")) {
-            return ifdLogger;
-        } else {
-            return logger;
-        }
+        return Loggers.getLogger(parentLogger, "." + component);
```
Comment (mikemccand) [inline core/src/main/java/org/elasticsearch/common/lucene/LoggerInfoStream.java]: I think this change inadvertently slowed down indexing: see annot AZ at https://benchmarks-old.elastic.co/index.html#indexing

The problem is (confusingly!) this private method is called for every log message that Lucene's `IndexWriter` makes.  Previously, we invoked `Loggers.getLogger` just twice on creating `LoggerInfoStream` but now we are calling it for every log message.

**Pull request merged**